cmd/otel-allocator/README.md

```diff
@@ -4,6 +4,135 @@ The TargetAllocator is an optional separately deployed component of an OpenTelem
 distribute targets of the PrometheusReceiver on all deployed Collector instances. The release version matches the
 operator's most recent release as well.
 
+In essence, Prometheus Receiver configs are "stolen" and replaced with a http_sd_configs directive that points to the 
+Allocator, these are then loadbalanced/sharded to the collectors. In addition, the TargetAllocator can discover targets via Prometheus CRs (currently ServiceMonitor,
+PodMonitor) which it presents on the `/jobs` endpoint.
+
+# Usage
+The `targetAllocator:` controls the TargetAllocator general properties. Full API spec can be found here: [api.md#opentelemetrycollectorspectargetallocator](../../docs/api.md#opentelemetrycollectorspectargetallocator)
+
+A basic example that deploys, but does nothing.
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: collector-with-ta
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+  config: |
+    receivers:
+      prometheus:
+        config:
+          scrape_configs:
+          - job_name: 'otel-collector'
+
+    exporters:
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [prometheus]
+          processors: []
+          exporters: [logging]
+```
+
+## PrometheusCR specifics
+TargetAllocator discovery of PrometheusCRs can be turned on with
+```yaml
+  targetAllocator:
+    enabled: true
+    prometheusCR:
+      enabled: true
+```
+The CRs can be filtered by labels as documented here: [api.md#opentelemetrycollectorspectargetallocatorprometheuscr](../../docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr)
+
+The prometheus receiver in the deployed collector also has to know where the Allocator service exists. This is done by a
+OpenTelemetry Collector operator specific config.
+```yaml
+  config: |
+    receivers:
+      prometheus:
+        config:
+          scrape_configs:
+          - job_name: 'otel-collector'
+        target_allocator:
+          endpoint: http://my-targetallocator-service
+          interval: 30s
+          collector_id: collector-1
+```
+Upstream documentation here: [Prometheusreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver#opentelemetry-operator)
+
+The TargetAllocator service is named based on the OpenTelemetryCollector CR name. `collector_id` should be unique per
+collector instance, such as the pod name. This can be passed down to the pod with env vars and replaced in the config
+with `${ENV}` notation.
```

Comment:
I'm unsure of this detail. It seems to work, but I don't know Why. Perhaps someone with more understanding can confirm this is indeed correct?

Reply:
You could replace this to be `${POD_NAME}` (it's automatically supplied by the operator [here](https://github.com/open-telemetry/opentelemetry-operator/blob/6d3da4219a9027aa49728d14036b84d47071b347/pkg/collector/container.go#L88))

Reply:
Updated